salt/modules/systemd_service.py

```diff
@@ -1346,3 +1346,58 @@ def execs(root=None):
             continue
         ret[service] = data['ExecStart']['path']
     return ret
+
+
+def firstboot(locale=None, locale_message=None, keymap=None,
+              timezone=None, hostname=None, machine_id=None,
+              root=None):
+    '''
+    .. versionadded:: TBD
+
+    Call systemd-firstboot to configure basic settings of the system
+
+    locale
+        Set primary locale (LANG=)
+
+    locale_message
+        Set message locale (LC_MESSAGES=)
+
+    keymap
+        Set keymap
+
+    timezone
+        Set timezone
+
+    hostname
+        Set host name
+
+    machine_id
+        Set machine ID
+
+    root
+        Operate on an alternative filesystem root
+
+    CLI Example:
+
+        salt '*' service.firstboot keymap=jp locale=en_US.UTF-8
+
+    '''
+    cmd = ['systemd-firstboot']
+    parameters = [('locale', locale),
+                  ('locale-message', locale_message),
+                  ('keymap', keymap),
```

Comment:
It would appear that keymap is not available on all systems - at least ```
CentOS Linux release 7.6.1810 (Core)
[vagrant@localhost ~]$ systemd-firstboot --keymap
systemd-firstboot: unrecognized option '--keymap'
```

Do you think it would make sense to provide a different message there, or should we just let the `unrecognized option` error bubble up?

Reply:
Ah good point!

IMHO is ok to let it go up. I will show this error message: "systemd-firstboot error: systemd-firstboot: unrecognized option '--keymap'". I think that is informative.